tests/testthat/test-example.R

```diff
@@ -505,3 +505,33 @@ test_that("fetch model size", {
   mod$set_index(c(3, 5, 7))
   expect_equal(mod$n_state(), 10)
 })
+
+
+test_that("resample", {
+  res <- dust_example("variable")
+  obj <- res$new(list(len = 5), 0, 7, seed = 1L)
+  m <- matrix(as.numeric(1:35), 5, 7)
+  obj$set_state(m)
+  rng <- dust_rng$new(obj$rng_state(TRUE))
+  u <- rng$unif_rand(1)
+  w <- runif(obj$n_particles())
```

Comment:
Can we try a couple of boundary cases on weights? e.g. only one non-zero weight, all zero-weights (should error)

Reply:
So, all-zero now just doesn't crash as we will cope with that in the filter itself (the situation where that happens the average ll comes out as zero and we stop). I've added a check for at least non-crashing behaviour now